spec/system/candidate_interface/candidate_existing_user_with_course_params_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+RSpec.describe 'An existing candidate arriving from Find with a course and provider code' do
+  include CourseOptionHelpers
+  scenario 'retaining their course selection through the sign up process' do
+    given_the_pilot_is_open
+    and_confirm_course_choice_from_find_is_activated
+    and_i_an_existing_candidate_on_apply
+    and_i_have_less_than_3_application_options
+    and_the_course_i_selected_only_has_one_site
+
+    when_i_arrive_at_the_sign_up_page_with_course_params_with_one_site
+    and_i_submit_my_email_address
+    and_click_on_the_magic_link
+    then_i_should_see_the_courses_review_page
+    and_i_should_see_the_course_name_and_code
+    and_i_should_see_the_site
+    and_my_find_from_id_course_should_be_set_to_nil
```

Comment:
Nit: doesn't feel like we should test for implementation details in a system spec, but the alternative in this particular case would be to duplicate a bunch of our assertions. We'd need to go through the whole process again, and make sure we don't get another offer to add the course. So I'm ok with this!